{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d623b919-954d-4022-aa57-e3b6ccc60c2d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from sklearn import preprocessing\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "664092ad-8f16-4494-9a6d-17f64b65c61c",
   "metadata": {},
   "source": [
    "# DATA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "dcfad844-2370-42a8-80d5-c4ac1e016dfe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#OTU ID</th>\n",
       "      <th>ITS-Ctrl-neg-extraction-Racine</th>\n",
       "      <th>ITS-Ctrl-neg-extraction-sol</th>\n",
       "      <th>ITS-Racine-AB1</th>\n",
       "      <th>ITS-Racine-AB11</th>\n",
       "      <th>ITS-Racine-AB12</th>\n",
       "      <th>ITS-Racine-FORE-001</th>\n",
       "      <th>ITS-Racine-FORE-004</th>\n",
       "      <th>ITS-Racine-FORE-007</th>\n",
       "      <th>ITS-Racine-FP2-3-5</th>\n",
       "      <th>...</th>\n",
       "      <th>taxonomy</th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Taxon Level</th>\n",
       "      <th>Trophic Mode</th>\n",
       "      <th>Guild</th>\n",
       "      <th>Growth Morphology</th>\n",
       "      <th>Trait</th>\n",
       "      <th>Confidence Ranking</th>\n",
       "      <th>Notes</th>\n",
       "      <th>Citation/Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3c9e177cc2b58b12790d1513b000757f</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>43774.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13581.0</td>\n",
       "      <td>...</td>\n",
       "      <td>k__Fungi; p__Ascomycota; c__Eurotiomycetes; o_...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Unassigned</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8a39bfe44d4064a5f6408bdfaa452f6f</td>\n",
       "      <td>120.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1926.0</td>\n",
       "      <td>1715.0</td>\n",
       "      <td>5027.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>19759.0</td>\n",
       "      <td>19251.0</td>\n",
       "      <td>6269.0</td>\n",
       "      <td>...</td>\n",
       "      <td>k__Fungi; p__Ascomycota; c__Sordariomycetes; o...</td>\n",
       "      <td>Fusarium</td>\n",
       "      <td>13</td>\n",
       "      <td>Pathotroph-Saprotroph-Symbiotroph</td>\n",
       "      <td>Animal Pathogen-Endophyte-Lichen Parasite-Plan...</td>\n",
       "      <td>Microfungus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Possible</td>\n",
       "      <td>Likely opportunistic human pathogen (Irinyi et...</td>\n",
       "      <td>b'Duncan CG, Eslyn WE. 1966. Mycologia 58:642-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>045a5e0e18a18f61c0ff982e9974e832</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>567.0</td>\n",
       "      <td>4858.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>113.0</td>\n",
       "      <td>924.0</td>\n",
       "      <td>...</td>\n",
       "      <td>k__Fungi</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Unassigned</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>b94ceff0b3bb4c03c4803a42d66e68fe</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>156.0</td>\n",
       "      <td>212.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>120.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>k__Fungi</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Unassigned</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9c364756b2f0dfe227401cef858935f6</td>\n",
       "      <td>21.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13663.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8447.0</td>\n",
       "      <td>2454.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>k__Fungi; p__Basidiomycota; c__Agaricomycetes;...</td>\n",
       "      <td>Thelephoraceae</td>\n",
       "      <td>9</td>\n",
       "      <td>Saprotroph-Symbiotroph</td>\n",
       "      <td>Ectomycorrhizal-Undefined Saprotroph</td>\n",
       "      <td>Clavarioid</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Probable</td>\n",
       "      <td>On the ground, on soil, a few on wood. At leas...</td>\n",
       "      <td>b'Cannon PF, Kirk PM. 2007. Fungal Families of...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            #OTU ID  ITS-Ctrl-neg-extraction-Racine  \\\n",
       "0  3c9e177cc2b58b12790d1513b000757f                             0.0   \n",
       "1  8a39bfe44d4064a5f6408bdfaa452f6f                           120.0   \n",
       "2  045a5e0e18a18f61c0ff982e9974e832                             8.0   \n",
       "3  b94ceff0b3bb4c03c4803a42d66e68fe                             0.0   \n",
       "4  9c364756b2f0dfe227401cef858935f6                            21.0   \n",
       "\n",
       "   ITS-Ctrl-neg-extraction-sol  ITS-Racine-AB1  ITS-Racine-AB11  \\\n",
       "0                          4.0         43774.0             30.0   \n",
       "1                          0.0          1926.0           1715.0   \n",
       "2                          0.0             0.0            567.0   \n",
       "3                          0.0             0.0            156.0   \n",
       "4                          0.0             0.0          13663.0   \n",
       "\n",
       "   ITS-Racine-AB12  ITS-Racine-FORE-001  ITS-Racine-FORE-004  \\\n",
       "0              3.0                  0.0                  0.0   \n",
       "1           5027.0                  0.0              19759.0   \n",
       "2           4858.0                 52.0                 63.0   \n",
       "3            212.0                  0.0                 19.0   \n",
       "4              0.0               8447.0               2454.0   \n",
       "\n",
       "   ITS-Racine-FORE-007  ITS-Racine-FP2-3-5  ...  \\\n",
       "0                  0.0             13581.0  ...   \n",
       "1              19251.0              6269.0  ...   \n",
       "2                113.0               924.0  ...   \n",
       "3                120.0                 0.0  ...   \n",
       "4                  0.0                 0.0  ...   \n",
       "\n",
       "                                            taxonomy           Taxon  \\\n",
       "0  k__Fungi; p__Ascomycota; c__Eurotiomycetes; o_...               -   \n",
       "1  k__Fungi; p__Ascomycota; c__Sordariomycetes; o...        Fusarium   \n",
       "2                                           k__Fungi               -   \n",
       "3                                           k__Fungi               -   \n",
       "4  k__Fungi; p__Basidiomycota; c__Agaricomycetes;...  Thelephoraceae   \n",
       "\n",
       "   Taxon Level                       Trophic Mode  \\\n",
       "0            -                                  -   \n",
       "1           13  Pathotroph-Saprotroph-Symbiotroph   \n",
       "2            -                                  -   \n",
       "3            -                                  -   \n",
       "4            9             Saprotroph-Symbiotroph   \n",
       "\n",
       "                                               Guild  Growth Morphology  \\\n",
       "0                                                  -                  -   \n",
       "1  Animal Pathogen-Endophyte-Lichen Parasite-Plan...        Microfungus   \n",
       "2                                                  -                  -   \n",
       "3                                                  -                  -   \n",
       "4               Ectomycorrhizal-Undefined Saprotroph         Clavarioid   \n",
       "\n",
       "   Trait  Confidence Ranking  \\\n",
       "0      -                   -   \n",
       "1    NaN            Possible   \n",
       "2      -                   -   \n",
       "3      -                   -   \n",
       "4    NaN            Probable   \n",
       "\n",
       "                                               Notes  \\\n",
       "0                                         Unassigned   \n",
       "1  Likely opportunistic human pathogen (Irinyi et...   \n",
       "2                                         Unassigned   \n",
       "3                                         Unassigned   \n",
       "4  On the ground, on soil, a few on wood. At leas...   \n",
       "\n",
       "                                     Citation/Source  \n",
       "0                                                  -  \n",
       "1  b'Duncan CG, Eslyn WE. 1966. Mycologia 58:642-...  \n",
       "2                                                  -  \n",
       "3                                                  -  \n",
       "4  b'Cannon PF, Kirk PM. 2007. Fungal Families of...  \n",
       "\n",
       "[5 rows x 49 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df = pd.read_csv(\"/mnt/d/NRCan/2022_MicrobiomeQuebecMexico/DATA/ITS/Full/Peuplier_REPRISE_ITS_oct2022.ASV_table_rarefied_11982_dnNA.guilds.txt\", sep='\\t', header=0)\n",
    "df = pd.read_csv(\"/mnt/d/NRCan/2022_MicrobiomeAspenGreenhouse/DATA/PeuplierGH_jan2024/ITS/all/PeuplierGH_ITS_jan2024.ASV_table_norarefaction_dnNA.guilds.txt\", sep='\\t', header=0)\n",
    "\n",
    "#df = df.set_index('#OTU ID')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "de36e3a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['#OTU ID', 'ITS-Ctrl-neg-extraction-Racine',\n",
       "       'ITS-Ctrl-neg-extraction-sol', 'ITS-Racine-AB1', 'ITS-Racine-AB11',\n",
       "       'ITS-Racine-AB12', 'ITS-Racine-FORE-001', 'ITS-Racine-FORE-004',\n",
       "       'ITS-Racine-FORE-007', 'ITS-Racine-FP2-3-5', 'ITS-Racine-FP2-5-11',\n",
       "       'ITS-Racine-FP2-5-17', 'ITS-Racine-SP-4', 'ITS-Racine-SP-45',\n",
       "       'ITS-Racine-SP-56', 'ITS-Rhizo-AB1', 'ITS-Rhizo-AB11',\n",
       "       'ITS-Rhizo-AB12', 'ITS-Rhizo-FORE-001', 'ITS-Rhizo-FORE-004',\n",
       "       'ITS-Rhizo-FORE-007', 'ITS-Rhizo-FP2-3-5', 'ITS-Rhizo-FP2-5-11',\n",
       "       'ITS-Rhizo-FP2-5-17', 'ITS-Rhizo-SP-4', 'ITS-Rhizo-SP-45',\n",
       "       'ITS-Rhizo-SP-56', 'ITS-sol-AB1', 'ITS-sol-AB11', 'ITS-sol-AB12',\n",
       "       'ITS-sol-FORE-001', 'ITS-sol-FORE-004', 'ITS-sol-FORE-007',\n",
       "       'ITS-sol-FP2-3-5', 'ITS-sol-FP2-5-11', 'ITS-sol-FP2-5-17',\n",
       "       'ITS-sol-SP-4', 'ITS-sol-SP-45', 'ITS-sol-SP-56', 'taxonomy',\n",
       "       'Taxon', 'Taxon Level', 'Trophic Mode', 'Guild',\n",
       "       'Growth Morphology', 'Trait', 'Confidence Ranking', 'Notes',\n",
       "       'Citation/Source'], dtype=object)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns.values"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dcf46b4a-5ec5-422a-bd07-e01bc9af1bef",
   "metadata": {},
   "source": [
    "## Abundance and taxonomy tables "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "0968eec0-e9f2-4587-9b7e-603aa8645e86",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1130, 10)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxonomy</th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Taxon Level</th>\n",
       "      <th>Trophic Mode</th>\n",
       "      <th>Guild</th>\n",
       "      <th>Growth Morphology</th>\n",
       "      <th>Trait</th>\n",
       "      <th>Confidence Ranking</th>\n",
       "      <th>Notes</th>\n",
       "      <th>Citation/Source</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#OTU ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3c9e177cc2b58b12790d1513b000757f</th>\n",
       "      <td>k__Fungi; p__Ascomycota; c__Eurotiomycetes; o_...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Unassigned</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8a39bfe44d4064a5f6408bdfaa452f6f</th>\n",
       "      <td>k__Fungi; p__Ascomycota; c__Sordariomycetes; o...</td>\n",
       "      <td>Fusarium</td>\n",
       "      <td>13</td>\n",
       "      <td>Pathotroph-Saprotroph-Symbiotroph</td>\n",
       "      <td>Animal Pathogen-Endophyte-Lichen Parasite-Plan...</td>\n",
       "      <td>Microfungus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Possible</td>\n",
       "      <td>Likely opportunistic human pathogen (Irinyi et...</td>\n",
       "      <td>b'Duncan CG, Eslyn WE. 1966. Mycologia 58:642-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>045a5e0e18a18f61c0ff982e9974e832</th>\n",
       "      <td>k__Fungi</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Unassigned</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b94ceff0b3bb4c03c4803a42d66e68fe</th>\n",
       "      <td>k__Fungi</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Unassigned</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9c364756b2f0dfe227401cef858935f6</th>\n",
       "      <td>k__Fungi; p__Basidiomycota; c__Agaricomycetes;...</td>\n",
       "      <td>Thelephoraceae</td>\n",
       "      <td>9</td>\n",
       "      <td>Saprotroph-Symbiotroph</td>\n",
       "      <td>Ectomycorrhizal-Undefined Saprotroph</td>\n",
       "      <td>Clavarioid</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Probable</td>\n",
       "      <td>On the ground, on soil, a few on wood. At leas...</td>\n",
       "      <td>b'Cannon PF, Kirk PM. 2007. Fungal Families of...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                           taxonomy  \\\n",
       "#OTU ID                                                                               \n",
       "3c9e177cc2b58b12790d1513b000757f  k__Fungi; p__Ascomycota; c__Eurotiomycetes; o_...   \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f  k__Fungi; p__Ascomycota; c__Sordariomycetes; o...   \n",
       "045a5e0e18a18f61c0ff982e9974e832                                           k__Fungi   \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe                                           k__Fungi   \n",
       "9c364756b2f0dfe227401cef858935f6  k__Fungi; p__Basidiomycota; c__Agaricomycetes;...   \n",
       "\n",
       "                                           Taxon Taxon Level  \\\n",
       "#OTU ID                                                        \n",
       "3c9e177cc2b58b12790d1513b000757f               -           -   \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f        Fusarium          13   \n",
       "045a5e0e18a18f61c0ff982e9974e832               -           -   \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe               -           -   \n",
       "9c364756b2f0dfe227401cef858935f6  Thelephoraceae           9   \n",
       "\n",
       "                                                       Trophic Mode  \\\n",
       "#OTU ID                                                               \n",
       "3c9e177cc2b58b12790d1513b000757f                                  -   \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f  Pathotroph-Saprotroph-Symbiotroph   \n",
       "045a5e0e18a18f61c0ff982e9974e832                                  -   \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe                                  -   \n",
       "9c364756b2f0dfe227401cef858935f6             Saprotroph-Symbiotroph   \n",
       "\n",
       "                                                                              Guild  \\\n",
       "#OTU ID                                                                               \n",
       "3c9e177cc2b58b12790d1513b000757f                                                  -   \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f  Animal Pathogen-Endophyte-Lichen Parasite-Plan...   \n",
       "045a5e0e18a18f61c0ff982e9974e832                                                  -   \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe                                                  -   \n",
       "9c364756b2f0dfe227401cef858935f6               Ectomycorrhizal-Undefined Saprotroph   \n",
       "\n",
       "                                 Growth Morphology Trait Confidence Ranking  \\\n",
       "#OTU ID                                                                       \n",
       "3c9e177cc2b58b12790d1513b000757f                 -     -                  -   \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f       Microfungus   NaN           Possible   \n",
       "045a5e0e18a18f61c0ff982e9974e832                 -     -                  -   \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe                 -     -                  -   \n",
       "9c364756b2f0dfe227401cef858935f6        Clavarioid   NaN           Probable   \n",
       "\n",
       "                                                                              Notes  \\\n",
       "#OTU ID                                                                               \n",
       "3c9e177cc2b58b12790d1513b000757f                                         Unassigned   \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f  Likely opportunistic human pathogen (Irinyi et...   \n",
       "045a5e0e18a18f61c0ff982e9974e832                                         Unassigned   \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe                                         Unassigned   \n",
       "9c364756b2f0dfe227401cef858935f6  On the ground, on soil, a few on wood. At leas...   \n",
       "\n",
       "                                                                    Citation/Source  \n",
       "#OTU ID                                                                              \n",
       "3c9e177cc2b58b12790d1513b000757f                                                  -  \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f  b'Duncan CG, Eslyn WE. 1966. Mycologia 58:642-...  \n",
       "045a5e0e18a18f61c0ff982e9974e832                                                  -  \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe                                                  -  \n",
       "9c364756b2f0dfe227401cef858935f6  b'Cannon PF, Kirk PM. 2007. Fungal Families of...  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "columns_its = [ele for ele in df.columns if 'ITS' in ele]\n",
    "columns_tax = [ele for ele in df.columns if 'ITS' not in ele]\n",
    "df_abu = df[columns_its]\n",
    "df_abu = df_abu.set_index(df['#OTU ID'])\n",
    "df_tax = df[columns_tax]\n",
    "df_tax = df_tax.set_index('#OTU ID')\n",
    "print(df_tax.shape)\n",
    "df_tax.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "700fe347",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1130, 1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>group</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#OTU ID</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3c9e177cc2b58b12790d1513b000757f</th>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8a39bfe44d4064a5f6408bdfaa452f6f</th>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>045a5e0e18a18f61c0ff982e9974e832</th>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b94ceff0b3bb4c03c4803a42d66e68fe</th>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9c364756b2f0dfe227401cef858935f6</th>\n",
       "      <td>Mixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dcad6acf788092fea97a22f2a6a5fb31</th>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b6b15037ee92cd7edaa6f3f6d3eeb6d0</th>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0432b9b1f641c119c2cfcb80d296ad60</th>\n",
       "      <td>Undefined Saprotroph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2400bdb0b04096c7f094639ce879660a</th>\n",
       "      <td>Plant Pathogen</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fe6e23bbbe96e05ca1e0bf96db68b8d5</th>\n",
       "      <td>Other</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1130 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 group\n",
       "#OTU ID                                               \n",
       "3c9e177cc2b58b12790d1513b000757f                 Other\n",
       "8a39bfe44d4064a5f6408bdfaa452f6f                 Other\n",
       "045a5e0e18a18f61c0ff982e9974e832                 Other\n",
       "b94ceff0b3bb4c03c4803a42d66e68fe                 Other\n",
       "9c364756b2f0dfe227401cef858935f6                 Mixed\n",
       "...                                                ...\n",
       "dcad6acf788092fea97a22f2a6a5fb31                 Other\n",
       "b6b15037ee92cd7edaa6f3f6d3eeb6d0                 Other\n",
       "0432b9b1f641c119c2cfcb80d296ad60  Undefined Saprotroph\n",
       "2400bdb0b04096c7f094639ce879660a        Plant Pathogen\n",
       "fe6e23bbbe96e05ca1e0bf96db68b8d5                 Other\n",
       "\n",
       "[1130 rows x 1 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_tax['group'] = df_tax['Guild'].fillna('Other')\n",
    "df_tax['group'] = df_tax['group'].apply(lambda x: x if x != \"-\" else \"Other\")\n",
    "df_tax['group'] = df_tax.apply(lambda x: x['group'] if x['Confidence Ranking'] in [\"Probable\",\"Highly Probable\"] else \"Other\", axis=1)\n",
    "df_tax['group'] = df_tax['group'].apply(lambda x: 'Mixed' if '-' in str(x) else str(x))\n",
    "df_tax.head()\n",
    "\n",
    "df_guilds = df_tax[['group']]\n",
    "print(df_guilds.shape)\n",
    "df_guilds"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26936d91-b02e-4e45-96d5-48b5d36f622a",
   "metadata": {},
   "source": [
    "## Subset soil samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "ad6ffa5b-6a96-44ab-9e67-a8ed9598d7de",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1130, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ITS-sol-AB1</th>\n",
       "      <th>ITS-sol-AB11</th>\n",
       "      <th>ITS-sol-AB12</th>\n",
       "      <th>ITS-sol-FORE-001</th>\n",
       "      <th>ITS-sol-FORE-004</th>\n",
       "      <th>ITS-sol-FORE-007</th>\n",
       "      <th>ITS-sol-FP2-3-5</th>\n",
       "      <th>ITS-sol-FP2-5-11</th>\n",
       "      <th>ITS-sol-FP2-5-17</th>\n",
       "      <th>ITS-sol-SP-4</th>\n",
       "      <th>ITS-sol-SP-45</th>\n",
       "      <th>ITS-sol-SP-56</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#OTU ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3c9e177cc2b58b12790d1513b000757f</th>\n",
       "      <td>12661.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>291.0</td>\n",
       "      <td>28987.0</td>\n",
       "      <td>16454.0</td>\n",
       "      <td>12949.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16181.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8a39bfe44d4064a5f6408bdfaa452f6f</th>\n",
       "      <td>1981.0</td>\n",
       "      <td>4914.0</td>\n",
       "      <td>2163.0</td>\n",
       "      <td>204.0</td>\n",
       "      <td>1480.0</td>\n",
       "      <td>4854.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>3040.0</td>\n",
       "      <td>2569.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>701.0</td>\n",
       "      <td>476.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>045a5e0e18a18f61c0ff982e9974e832</th>\n",
       "      <td>382.0</td>\n",
       "      <td>6095.0</td>\n",
       "      <td>6193.0</td>\n",
       "      <td>4083.0</td>\n",
       "      <td>960.0</td>\n",
       "      <td>428.0</td>\n",
       "      <td>1399.0</td>\n",
       "      <td>4867.0</td>\n",
       "      <td>1077.0</td>\n",
       "      <td>3727.0</td>\n",
       "      <td>363.0</td>\n",
       "      <td>237.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b94ceff0b3bb4c03c4803a42d66e68fe</th>\n",
       "      <td>12.0</td>\n",
       "      <td>4148.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>136.0</td>\n",
       "      <td>629.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>255.0</td>\n",
       "      <td>21934.0</td>\n",
       "      <td>2722.0</td>\n",
       "      <td>12759.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9c364756b2f0dfe227401cef858935f6</th>\n",
       "      <td>0.0</td>\n",
       "      <td>4402.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4723.0</td>\n",
       "      <td>7837.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1364.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  ITS-sol-AB1  ITS-sol-AB11  ITS-sol-AB12  \\\n",
       "#OTU ID                                                                     \n",
       "3c9e177cc2b58b12790d1513b000757f      12661.0           0.0           0.0   \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f       1981.0        4914.0        2163.0   \n",
       "045a5e0e18a18f61c0ff982e9974e832        382.0        6095.0        6193.0   \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe         12.0        4148.0          87.0   \n",
       "9c364756b2f0dfe227401cef858935f6          0.0        4402.0           0.0   \n",
       "\n",
       "                                  ITS-sol-FORE-001  ITS-sol-FORE-004  \\\n",
       "#OTU ID                                                                \n",
       "3c9e177cc2b58b12790d1513b000757f               0.0               0.0   \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f             204.0            1480.0   \n",
       "045a5e0e18a18f61c0ff982e9974e832            4083.0             960.0   \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe              20.0             136.0   \n",
       "9c364756b2f0dfe227401cef858935f6            4723.0            7837.0   \n",
       "\n",
       "                                  ITS-sol-FORE-007  ITS-sol-FP2-3-5  \\\n",
       "#OTU ID                                                               \n",
       "3c9e177cc2b58b12790d1513b000757f               0.0            291.0   \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f            4854.0            472.0   \n",
       "045a5e0e18a18f61c0ff982e9974e832             428.0           1399.0   \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe             629.0              0.0   \n",
       "9c364756b2f0dfe227401cef858935f6               0.0              0.0   \n",
       "\n",
       "                                  ITS-sol-FP2-5-11  ITS-sol-FP2-5-17  \\\n",
       "#OTU ID                                                                \n",
       "3c9e177cc2b58b12790d1513b000757f           28987.0           16454.0   \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f            3040.0            2569.0   \n",
       "045a5e0e18a18f61c0ff982e9974e832            4867.0            1077.0   \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe             255.0           21934.0   \n",
       "9c364756b2f0dfe227401cef858935f6               0.0               0.0   \n",
       "\n",
       "                                  ITS-sol-SP-4  ITS-sol-SP-45  ITS-sol-SP-56  \n",
       "#OTU ID                                                                       \n",
       "3c9e177cc2b58b12790d1513b000757f       12949.0            0.0        16181.0  \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f         300.0          701.0          476.0  \n",
       "045a5e0e18a18f61c0ff982e9974e832        3727.0          363.0          237.0  \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe        2722.0        12759.0            0.0  \n",
       "9c364756b2f0dfe227401cef858935f6           0.0            0.0         1364.0  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "columns_soil = [ele for ele in df if 'sol' in ele and 'Ctrl' not in ele]\n",
    "df_abu_soil = df_abu[columns_soil]\n",
    "print(df_abu_soil.shape)\n",
    "df_abu_soil.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7095e9ba-d9b2-46e6-a972-efb806665152",
   "metadata": {},
   "source": [
    "## Metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "8b714daf-ddd2-4c15-bbed-c5001b4dcd4a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SampleID</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Sample type</th>\n",
       "      <th>Population</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ITS-Phyllo-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>phyllosphere</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ITS-Phyllo-AB11</td>\n",
       "      <td>AB11</td>\n",
       "      <td>phyllosphere</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ITS-Phyllo-AB12</td>\n",
       "      <td>AB12</td>\n",
       "      <td>phyllosphere</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          SampleID Genotype   Sample type Population Cluster  Unnamed: 5  \\\n",
       "0   ITS-Phyllo-AB1      AB1  phyllosphere       NENA      AB         NaN   \n",
       "1  ITS-Phyllo-AB11     AB11  phyllosphere       NENA      AB         NaN   \n",
       "2  ITS-Phyllo-AB12     AB12  phyllosphere       NENA      AB         NaN   \n",
       "\n",
       "   Unnamed: 6  \n",
       "0         NaN  \n",
       "1         NaN  \n",
       "2         NaN  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta = pd.read_csv(\"/mnt/d/NRCan/2022_MicrobiomeAspenGreenhouse/DATA/PeuplierGH_jan2024/metadata_ITS.txt\", sep=\"\\t\", header=0)\n",
    "df_meta.head(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a7350e95-7671-4bd7-86a0-6fca26035af1",
   "metadata": {},
   "source": [
    "# Sum abundances across guilds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "10011620-52d8-4db4-935e-02124fc75b4d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1130, 1)\n",
      "(1130, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ITS-sol-AB1</th>\n",
       "      <th>ITS-sol-AB11</th>\n",
       "      <th>ITS-sol-AB12</th>\n",
       "      <th>ITS-sol-FORE-001</th>\n",
       "      <th>ITS-sol-FORE-004</th>\n",
       "      <th>ITS-sol-FORE-007</th>\n",
       "      <th>ITS-sol-FP2-3-5</th>\n",
       "      <th>ITS-sol-FP2-5-11</th>\n",
       "      <th>ITS-sol-FP2-5-17</th>\n",
       "      <th>ITS-sol-SP-4</th>\n",
       "      <th>ITS-sol-SP-45</th>\n",
       "      <th>ITS-sol-SP-56</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#OTU ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3c9e177cc2b58b12790d1513b000757f</th>\n",
       "      <td>12661.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>291.0</td>\n",
       "      <td>28987.0</td>\n",
       "      <td>16454.0</td>\n",
       "      <td>12949.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16181.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8a39bfe44d4064a5f6408bdfaa452f6f</th>\n",
       "      <td>1981.0</td>\n",
       "      <td>4914.0</td>\n",
       "      <td>2163.0</td>\n",
       "      <td>204.0</td>\n",
       "      <td>1480.0</td>\n",
       "      <td>4854.0</td>\n",
       "      <td>472.0</td>\n",
       "      <td>3040.0</td>\n",
       "      <td>2569.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>701.0</td>\n",
       "      <td>476.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>045a5e0e18a18f61c0ff982e9974e832</th>\n",
       "      <td>382.0</td>\n",
       "      <td>6095.0</td>\n",
       "      <td>6193.0</td>\n",
       "      <td>4083.0</td>\n",
       "      <td>960.0</td>\n",
       "      <td>428.0</td>\n",
       "      <td>1399.0</td>\n",
       "      <td>4867.0</td>\n",
       "      <td>1077.0</td>\n",
       "      <td>3727.0</td>\n",
       "      <td>363.0</td>\n",
       "      <td>237.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b94ceff0b3bb4c03c4803a42d66e68fe</th>\n",
       "      <td>12.0</td>\n",
       "      <td>4148.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>136.0</td>\n",
       "      <td>629.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>255.0</td>\n",
       "      <td>21934.0</td>\n",
       "      <td>2722.0</td>\n",
       "      <td>12759.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9c364756b2f0dfe227401cef858935f6</th>\n",
       "      <td>0.0</td>\n",
       "      <td>4402.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4723.0</td>\n",
       "      <td>7837.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1364.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  ITS-sol-AB1  ITS-sol-AB11  ITS-sol-AB12  \\\n",
       "#OTU ID                                                                     \n",
       "3c9e177cc2b58b12790d1513b000757f      12661.0           0.0           0.0   \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f       1981.0        4914.0        2163.0   \n",
       "045a5e0e18a18f61c0ff982e9974e832        382.0        6095.0        6193.0   \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe         12.0        4148.0          87.0   \n",
       "9c364756b2f0dfe227401cef858935f6          0.0        4402.0           0.0   \n",
       "\n",
       "                                  ITS-sol-FORE-001  ITS-sol-FORE-004  \\\n",
       "#OTU ID                                                                \n",
       "3c9e177cc2b58b12790d1513b000757f               0.0               0.0   \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f             204.0            1480.0   \n",
       "045a5e0e18a18f61c0ff982e9974e832            4083.0             960.0   \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe              20.0             136.0   \n",
       "9c364756b2f0dfe227401cef858935f6            4723.0            7837.0   \n",
       "\n",
       "                                  ITS-sol-FORE-007  ITS-sol-FP2-3-5  \\\n",
       "#OTU ID                                                               \n",
       "3c9e177cc2b58b12790d1513b000757f               0.0            291.0   \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f            4854.0            472.0   \n",
       "045a5e0e18a18f61c0ff982e9974e832             428.0           1399.0   \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe             629.0              0.0   \n",
       "9c364756b2f0dfe227401cef858935f6               0.0              0.0   \n",
       "\n",
       "                                  ITS-sol-FP2-5-11  ITS-sol-FP2-5-17  \\\n",
       "#OTU ID                                                                \n",
       "3c9e177cc2b58b12790d1513b000757f           28987.0           16454.0   \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f            3040.0            2569.0   \n",
       "045a5e0e18a18f61c0ff982e9974e832            4867.0            1077.0   \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe             255.0           21934.0   \n",
       "9c364756b2f0dfe227401cef858935f6               0.0               0.0   \n",
       "\n",
       "                                  ITS-sol-SP-4  ITS-sol-SP-45  ITS-sol-SP-56  \n",
       "#OTU ID                                                                       \n",
       "3c9e177cc2b58b12790d1513b000757f       12949.0            0.0        16181.0  \n",
       "8a39bfe44d4064a5f6408bdfaa452f6f         300.0          701.0          476.0  \n",
       "045a5e0e18a18f61c0ff982e9974e832        3727.0          363.0          237.0  \n",
       "b94ceff0b3bb4c03c4803a42d66e68fe        2722.0        12759.0            0.0  \n",
       "9c364756b2f0dfe227401cef858935f6           0.0            0.0         1364.0  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(df_guilds.shape)\n",
    "print(df_abu_soil.shape)\n",
    "df_abu_soil.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "7ddb5d1d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ITS-sol-AB1</th>\n",
       "      <th>ITS-sol-AB11</th>\n",
       "      <th>ITS-sol-AB12</th>\n",
       "      <th>ITS-sol-FORE-001</th>\n",
       "      <th>ITS-sol-FORE-004</th>\n",
       "      <th>ITS-sol-FORE-007</th>\n",
       "      <th>ITS-sol-FP2-3-5</th>\n",
       "      <th>ITS-sol-FP2-5-11</th>\n",
       "      <th>ITS-sol-FP2-5-17</th>\n",
       "      <th>ITS-sol-SP-4</th>\n",
       "      <th>ITS-sol-SP-45</th>\n",
       "      <th>ITS-sol-SP-56</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>group</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Animal Pathogen</th>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>112.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Dung Saprotroph</th>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>251.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ectomycorrhizal</th>\n",
       "      <td>5286.0</td>\n",
       "      <td>1801.0</td>\n",
       "      <td>14649.0</td>\n",
       "      <td>9707.0</td>\n",
       "      <td>9380.0</td>\n",
       "      <td>2232.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>111.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1659.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Endophyte</th>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ericoid Mycorrhizal</th>\n",
       "      <td>759.0</td>\n",
       "      <td>223.0</td>\n",
       "      <td>275.0</td>\n",
       "      <td>501.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>310.0</td>\n",
       "      <td>250.0</td>\n",
       "      <td>596.0</td>\n",
       "      <td>513.0</td>\n",
       "      <td>219.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>102.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     ITS-sol-AB1  ITS-sol-AB11  ITS-sol-AB12  \\\n",
       "group                                                          \n",
       "Animal Pathogen             14.0           0.0           0.0   \n",
       "Dung Saprotroph              7.0           0.0           2.0   \n",
       "Ectomycorrhizal           5286.0        1801.0       14649.0   \n",
       "Endophyte                   10.0           0.0          13.0   \n",
       "Ericoid Mycorrhizal        759.0         223.0         275.0   \n",
       "\n",
       "                     ITS-sol-FORE-001  ITS-sol-FORE-004  ITS-sol-FORE-007  \\\n",
       "group                                                                       \n",
       "Animal Pathogen                  36.0             112.0               0.0   \n",
       "Dung Saprotroph                   0.0               0.0               0.0   \n",
       "Ectomycorrhizal                9707.0            9380.0            2232.0   \n",
       "Endophyte                         3.0               0.0              20.0   \n",
       "Ericoid Mycorrhizal             501.0              65.0             310.0   \n",
       "\n",
       "                     ITS-sol-FP2-3-5  ITS-sol-FP2-5-11  ITS-sol-FP2-5-17  \\\n",
       "group                                                                      \n",
       "Animal Pathogen                 27.0               4.0               4.0   \n",
       "Dung Saprotroph                  0.0               0.0               0.0   \n",
       "Ectomycorrhizal                  0.0              63.0             111.0   \n",
       "Endophyte                        0.0              74.0               0.0   \n",
       "Ericoid Mycorrhizal            250.0             596.0             513.0   \n",
       "\n",
       "                     ITS-sol-SP-4  ITS-sol-SP-45  ITS-sol-SP-56  \n",
       "group                                                            \n",
       "Animal Pathogen               0.0            0.0            0.0  \n",
       "Dung Saprotroph             251.0           27.0           13.0  \n",
       "Ectomycorrhizal              28.0            0.0         1659.0  \n",
       "Endophyte                    31.0            0.0            6.0  \n",
       "Ericoid Mycorrhizal         219.0           96.0          102.0  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sum = df_abu_soil.groupby(df_guilds['group']).sum()\n",
    "df_sum.to_csv(\"guilds_02_read_greenhouse.csv\", sep=\",\", header=True, index=True)\n",
    "\n",
    "df_sum.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "691fb525",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>group</th>\n",
       "      <th>count</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Sample type</th>\n",
       "      <th>Population</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>long.name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ITS-sol-AB1</td>\n",
       "      <td>Animal Pathogen</td>\n",
       "      <td>14.0</td>\n",
       "      <td>ITS-sol-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>bulk soil</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.sol.AB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ITS-sol-AB1</td>\n",
       "      <td>Dung Saprotroph</td>\n",
       "      <td>7.0</td>\n",
       "      <td>ITS-sol-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>bulk soil</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.sol.AB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ITS-sol-AB1</td>\n",
       "      <td>Ectomycorrhizal</td>\n",
       "      <td>5286.0</td>\n",
       "      <td>ITS-sol-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>bulk soil</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.sol.AB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ITS-sol-AB1</td>\n",
       "      <td>Endophyte</td>\n",
       "      <td>10.0</td>\n",
       "      <td>ITS-sol-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>bulk soil</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.sol.AB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ITS-sol-AB1</td>\n",
       "      <td>Ericoid Mycorrhizal</td>\n",
       "      <td>759.0</td>\n",
       "      <td>ITS-sol-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>bulk soil</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.sol.AB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>ITS-sol-SP-56</td>\n",
       "      <td>Other</td>\n",
       "      <td>37975.0</td>\n",
       "      <td>ITS-sol-SP-56</td>\n",
       "      <td>SP-56</td>\n",
       "      <td>bulk soil</td>\n",
       "      <td>MX</td>\n",
       "      <td>SP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.sol.SP.56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>ITS-sol-SP-56</td>\n",
       "      <td>Plant Pathogen</td>\n",
       "      <td>29.0</td>\n",
       "      <td>ITS-sol-SP-56</td>\n",
       "      <td>SP-56</td>\n",
       "      <td>bulk soil</td>\n",
       "      <td>MX</td>\n",
       "      <td>SP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.sol.SP.56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>ITS-sol-SP-56</td>\n",
       "      <td>Soil Saprotroph</td>\n",
       "      <td>11.0</td>\n",
       "      <td>ITS-sol-SP-56</td>\n",
       "      <td>SP-56</td>\n",
       "      <td>bulk soil</td>\n",
       "      <td>MX</td>\n",
       "      <td>SP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.sol.SP.56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>ITS-sol-SP-56</td>\n",
       "      <td>Undefined Saprotroph</td>\n",
       "      <td>5345.0</td>\n",
       "      <td>ITS-sol-SP-56</td>\n",
       "      <td>SP-56</td>\n",
       "      <td>bulk soil</td>\n",
       "      <td>MX</td>\n",
       "      <td>SP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.sol.SP.56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>ITS-sol-SP-56</td>\n",
       "      <td>Wood Saprotroph</td>\n",
       "      <td>127.0</td>\n",
       "      <td>ITS-sol-SP-56</td>\n",
       "      <td>SP-56</td>\n",
       "      <td>bulk soil</td>\n",
       "      <td>MX</td>\n",
       "      <td>SP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.sol.SP.56</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>180 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            sample                 group    count       SampleID Genotype  \\\n",
       "0      ITS-sol-AB1       Animal Pathogen     14.0    ITS-sol-AB1      AB1   \n",
       "1      ITS-sol-AB1       Dung Saprotroph      7.0    ITS-sol-AB1      AB1   \n",
       "2      ITS-sol-AB1       Ectomycorrhizal   5286.0    ITS-sol-AB1      AB1   \n",
       "3      ITS-sol-AB1             Endophyte     10.0    ITS-sol-AB1      AB1   \n",
       "4      ITS-sol-AB1   Ericoid Mycorrhizal    759.0    ITS-sol-AB1      AB1   \n",
       "..             ...                   ...      ...            ...      ...   \n",
       "175  ITS-sol-SP-56                 Other  37975.0  ITS-sol-SP-56    SP-56   \n",
       "176  ITS-sol-SP-56        Plant Pathogen     29.0  ITS-sol-SP-56    SP-56   \n",
       "177  ITS-sol-SP-56       Soil Saprotroph     11.0  ITS-sol-SP-56    SP-56   \n",
       "178  ITS-sol-SP-56  Undefined Saprotroph   5345.0  ITS-sol-SP-56    SP-56   \n",
       "179  ITS-sol-SP-56       Wood Saprotroph    127.0  ITS-sol-SP-56    SP-56   \n",
       "\n",
       "    Sample type Population Cluster  Unnamed: 5  Unnamed: 6      long.name  \n",
       "0     bulk soil       NENA      AB         NaN         NaN    ITS.sol.AB1  \n",
       "1     bulk soil       NENA      AB         NaN         NaN    ITS.sol.AB1  \n",
       "2     bulk soil       NENA      AB         NaN         NaN    ITS.sol.AB1  \n",
       "3     bulk soil       NENA      AB         NaN         NaN    ITS.sol.AB1  \n",
       "4     bulk soil       NENA      AB         NaN         NaN    ITS.sol.AB1  \n",
       "..          ...        ...     ...         ...         ...            ...  \n",
       "175   bulk soil         MX      SP         NaN         NaN  ITS.sol.SP.56  \n",
       "176   bulk soil         MX      SP         NaN         NaN  ITS.sol.SP.56  \n",
       "177   bulk soil         MX      SP         NaN         NaN  ITS.sol.SP.56  \n",
       "178   bulk soil         MX      SP         NaN         NaN  ITS.sol.SP.56  \n",
       "179   bulk soil         MX      SP         NaN         NaN  ITS.sol.SP.56  \n",
       "\n",
       "[180 rows x 11 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dt_sum = df_sum.T\n",
    "stack_guilds = dt_sum.stack().reset_index().rename(columns={'level_0' : 'sample', 0 : 'count'})\n",
    "df_meta_summed = pd.merge(stack_guilds, df_meta, left_on = 'sample', right_on = 'SampleID', how='left')\n",
    "df_meta_summed['long.name'] = df_meta_summed['sample'].apply(lambda x: x.replace('-','.'))\n",
    "df_meta_summed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "509beb0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_meta_summed.to_csv(\"guilds_02_read_greenhouse_meta.csv\", sep=\",\", header=True, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fa01e1e6",
   "metadata": {},
   "source": [
    "# Rhizosphere"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "93efea7f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1130, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>group</th>\n",
       "      <th>count</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Sample type</th>\n",
       "      <th>Population</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>long.name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ITS-Rhizo-AB1</td>\n",
       "      <td>Animal Pathogen</td>\n",
       "      <td>47.0</td>\n",
       "      <td>ITS-Rhizo-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>rhizosphere</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.Rhizo.AB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ITS-Rhizo-AB1</td>\n",
       "      <td>Dung Saprotroph</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ITS-Rhizo-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>rhizosphere</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.Rhizo.AB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ITS-Rhizo-AB1</td>\n",
       "      <td>Ectomycorrhizal</td>\n",
       "      <td>3970.0</td>\n",
       "      <td>ITS-Rhizo-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>rhizosphere</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.Rhizo.AB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ITS-Rhizo-AB1</td>\n",
       "      <td>Endophyte</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ITS-Rhizo-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>rhizosphere</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.Rhizo.AB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ITS-Rhizo-AB1</td>\n",
       "      <td>Ericoid Mycorrhizal</td>\n",
       "      <td>778.0</td>\n",
       "      <td>ITS-Rhizo-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>rhizosphere</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.Rhizo.AB1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          sample                group   count       SampleID Genotype  \\\n",
       "0  ITS-Rhizo-AB1      Animal Pathogen    47.0  ITS-Rhizo-AB1      AB1   \n",
       "1  ITS-Rhizo-AB1      Dung Saprotroph     0.0  ITS-Rhizo-AB1      AB1   \n",
       "2  ITS-Rhizo-AB1      Ectomycorrhizal  3970.0  ITS-Rhizo-AB1      AB1   \n",
       "3  ITS-Rhizo-AB1            Endophyte     0.0  ITS-Rhizo-AB1      AB1   \n",
       "4  ITS-Rhizo-AB1  Ericoid Mycorrhizal   778.0  ITS-Rhizo-AB1      AB1   \n",
       "\n",
       "   Sample type Population Cluster  Unnamed: 5  Unnamed: 6      long.name  \n",
       "0  rhizosphere       NENA      AB         NaN         NaN  ITS.Rhizo.AB1  \n",
       "1  rhizosphere       NENA      AB         NaN         NaN  ITS.Rhizo.AB1  \n",
       "2  rhizosphere       NENA      AB         NaN         NaN  ITS.Rhizo.AB1  \n",
       "3  rhizosphere       NENA      AB         NaN         NaN  ITS.Rhizo.AB1  \n",
       "4  rhizosphere       NENA      AB         NaN         NaN  ITS.Rhizo.AB1  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Subsetting samples\n",
    "df.columns.values\n",
    "columns_sub = [ele for ele in df if 'Rhizo' in ele and 'Ctrl' not in ele]\n",
    "df_abu_sub = df_abu[columns_sub]\n",
    "print(df_abu_sub.shape)\n",
    "\n",
    "# Summing guilds\n",
    "df_sum = df_abu_sub.groupby(df_guilds['group']).sum()\n",
    "df_sum\n",
    "df_sum.to_csv(\"guilds_02_read_greenhouse_rhizo.csv\", sep=\",\", header=True, index=True)\n",
    "\n",
    "# Merging with metadata\n",
    "df_meta = pd.read_csv(\"/mnt/d/NRCan/2022_MicrobiomeAspenGreenhouse/DATA/PeuplierGH_jan2024/metadata_ITS.txt\", sep=\"\\t\", header=0)\n",
    "dt_sum = df_sum.T\n",
    "stack_guilds = dt_sum.stack().reset_index().rename(columns={'level_0' : 'sample', 0 : 'count'})\n",
    "df_meta_summed = pd.merge(stack_guilds, df_meta, left_on = 'sample', right_on = 'SampleID', how='left')\n",
    "df_meta_summed['long.name'] = df_meta_summed['sample'].apply(lambda x: x.replace('-','.'))\n",
    "df_meta_summed.to_csv(\"guilds_02_read_greenhouse_meta_rhizo.csv\", sep=\",\", header=True, index=False)\n",
    "\n",
    "df_meta_summed.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8372499b",
   "metadata": {},
   "source": [
    "# Roots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "087cb8cd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1130, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>group</th>\n",
       "      <th>count</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Sample type</th>\n",
       "      <th>Population</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>long.name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ITS-Racine-AB1</td>\n",
       "      <td>Animal Pathogen</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ITS-Racine-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>roots</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.Racine.AB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ITS-Racine-AB1</td>\n",
       "      <td>Dung Saprotroph</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ITS-Racine-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>roots</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.Racine.AB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ITS-Racine-AB1</td>\n",
       "      <td>Ectomycorrhizal</td>\n",
       "      <td>5833.0</td>\n",
       "      <td>ITS-Racine-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>roots</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.Racine.AB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ITS-Racine-AB1</td>\n",
       "      <td>Endophyte</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ITS-Racine-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>roots</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.Racine.AB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ITS-Racine-AB1</td>\n",
       "      <td>Ericoid Mycorrhizal</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ITS-Racine-AB1</td>\n",
       "      <td>AB1</td>\n",
       "      <td>roots</td>\n",
       "      <td>NENA</td>\n",
       "      <td>AB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ITS.Racine.AB1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           sample                group   count        SampleID Genotype  \\\n",
       "0  ITS-Racine-AB1      Animal Pathogen     0.0  ITS-Racine-AB1      AB1   \n",
       "1  ITS-Racine-AB1      Dung Saprotroph     0.0  ITS-Racine-AB1      AB1   \n",
       "2  ITS-Racine-AB1      Ectomycorrhizal  5833.0  ITS-Racine-AB1      AB1   \n",
       "3  ITS-Racine-AB1            Endophyte     0.0  ITS-Racine-AB1      AB1   \n",
       "4  ITS-Racine-AB1  Ericoid Mycorrhizal     5.0  ITS-Racine-AB1      AB1   \n",
       "\n",
       "  Sample type Population Cluster  Unnamed: 5  Unnamed: 6       long.name  \n",
       "0       roots       NENA      AB         NaN         NaN  ITS.Racine.AB1  \n",
       "1       roots       NENA      AB         NaN         NaN  ITS.Racine.AB1  \n",
       "2       roots       NENA      AB         NaN         NaN  ITS.Racine.AB1  \n",
       "3       roots       NENA      AB         NaN         NaN  ITS.Racine.AB1  \n",
       "4       roots       NENA      AB         NaN         NaN  ITS.Racine.AB1  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Subsetting samples\n",
    "df.columns.values\n",
    "columns_sub = [ele for ele in df if 'Racine' in ele and 'Ctrl' not in ele]\n",
    "df_abu_sub = df_abu[columns_sub]\n",
    "print(df_abu_sub.shape)\n",
    "\n",
    "# Summing guilds\n",
    "df_sum = df_abu_sub.groupby(df_guilds['group']).sum()\n",
    "df_sum.to_csv(\"guilds_02_read_greenhouse_roots.csv\", sep=\",\", header=True, index=True)\n",
    "\n",
    "# Merging with metadata\n",
    "df_meta = pd.read_csv(\"/mnt/d/NRCan/2022_MicrobiomeAspenGreenhouse/DATA/PeuplierGH_jan2024/metadata_ITS.txt\", sep=\"\\t\", header=0)\n",
    "dt_sum = df_sum.T\n",
    "stack_guilds = dt_sum.stack().reset_index().rename(columns={'level_0' : 'sample', 0 : 'count'})\n",
    "df_meta_summed = pd.merge(stack_guilds, df_meta, left_on = 'sample', right_on = 'SampleID', how='left')\n",
    "df_meta_summed['long.name'] = df_meta_summed['sample'].apply(lambda x: x.replace('-','.'))\n",
    "df_meta_summed.to_csv(\"guilds_02_read_greenhouse_meta_roots.csv\", sep=\",\", header=True, index=False)\n",
    "\n",
    "df_meta_summed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "59b23237",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1474ab17",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "253b3053",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "mbio",
   "language": "python",
   "name": "mbio"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
